scipy/stats/_morestats.py

```diff
@@ -3904,3 +3904,71 @@ def circstd(samples, high=2*pi, low=0, axis=None, nan_policy='propagate', *,
     if not normalize:
         res *= (high-low)/(2.*pi)  # [1] (2.3.14) w/ (2.3.7)
     return res
+
+
+@_axis_nan_policy_factory(
+    lambda x: x, result_to_tuple=lambda x: (x,), n_outputs=1
+)
+def directionalmean(samples, *, axis=0):
+    """
+    Computes the directional mean of a sample of vectors.
+
+    Serves as equivalent of the sample mean for directional data whose
+    magnitude is irrelevant, e. g. unit vectors. The vectorial data
+    whose directionals means are computed must be located in the last
+    axis.
+
+    Parameters
+    ----------
+    samples : array_like
+        Input array. Must be at least two-dimensional.
+    axis : int, optional
+        Axis along which directional means are computed. Default is 0.
+
+    Returns
+    -------
+    directionalmean : ndarray
+        Directional mean.
+
+    Notes
+    -----
+    This uses a definition of directional mean from [1]_.
+    Essentially, the calculation is as follows.
+
+    .. code-block:: python
+
+        mean=samples.mean()
+        directionalmean = mean/np.linalg.norm(mean)
```

Comment:
```suggestion
        mean=samples.mean(axis=0)
        directionalmean = mean/np.linalg.norm(mean)
```
In 1d statistics, I think it's OK to write these assuming the data is 1D, so we don't normally include `axis`. Here, the minimum dimensionality is 2D, so I think that we need to write the code for 2D. As is, the code would fail in 2D because the default is `axis=None`, so I think we need to include the `axis` of the first `mean`.